text/0000-macros-1.1.md

```diff
@@ -0,0 +1,494 @@
+
+  As mentioned before, however, the stable version of macros 2.0 is currently
+  quite far off, and the desire for features like custom derive are very real
+  today. The rationale behind this RFC is that the downsides are an acceptable
+  tradeoff from moving a significat portion of the nightly ecosystem onto stable
+  Rust.
+
+* This implementation is likely to be less performant than procedural macros
+  are today. Round tripping through strings isn't always a speedy operation,
+  especially for larger expantions. Strings, however, are a very small
+  implementation detail that's easy to see stabilized until the end of time.
+  Additionally, it's planned to extend the `TokenStream` API in the future to
+  allow more fine-grained transformations without having to round trip through
+  strings.
+
+* Users will still have an inferior experience to today's nightly macros
+  specifically with respect to compile times. The `syntex_syntax` crate takes
+  quite a few seconds to compile, and this would be required by any crate which
+  uses serde. To offset this, though, the `syntex_syntax` could be *massively*
+  stripped down as all it needs to do is parse struct declarations mostly. There
+  are likely many other various optimizations to compile time that can be
+  applied to ensure that it compiles quickly.
+
+# Alternatives
+[alternatives]: #alternatives
+
+* Wait for macros 2.0, but this likely comes with the high cost of postponing a
+  stable custom-derive experience on the time scale of years.
+
+* Don't stabilize `rustc_macro` as a new crate, but rather specify that
+  `#[rustc_macro_derive]` has a stable-ABI friendly signature. This does not
+  account, however, for the eventual planned introduction of the `rustc_macro`
+  crate and is significantly harder to write. The marginal benefit of being
+  slightly more flexible about how it's run likely isn't worth it.
+
+* The syntax for defining a macro may be different in the macros 2.0 world (e.g.
+  `pub macro foo` vs an attribute), that is it probably won't involve a function
+  attribute like `#[rustc_macro_derive]`. This interim system could possibly use
+  this syntax as well, but it's unclear whether we have a concrete enough idea
+  in mind to implement today.
+
+* Instead of passing around `&mut MacroContext` we could allow for storage of
+  compiler data structures in thread-local-storage. This would avoid threading
+  around an extra parameter and perhaps wouldn't lose too much flexibility.
+
+* Instead of `#![rustc_macro_crate]` macro crates could be loaded with `extern
+  crate foo` instead. The downsides of this are that the compiler would have to
+  generate an error if you import paths from it, so it doesn't behave like other
+  `extern crate` statements, and the compiler also doesn't know whether to look
+  for a target or host crate when it encounters such a definition. If both are
+  found it doesn't know which is the right to choose as well.
```

Comment:
I think this is the only real downside. Could `Cargo` handle this correctly by default?